pallets/avn-proxy/src/lib.rs

```diff
@@ -211,7 +233,7 @@ impl<T: Config> Pallet<T> {
 
     pub(crate) fn charge_fee(
         proof: &Proof<T::Signature, T::AccountId>,
-        payment_info: PaymentInfo<T::AccountId, BalanceOf<T>, T::Signature>,
+        payment_info: PaymentInfo<T::AccountId, BalanceOf<T>, T::Signature, T::Token>,
```

Comment:
can this be abused? meaning any ERC20 token can be used as a payment proof?

Reply:
This is a good question. If someone sends a transaction direct to chain, and decides to pay for it, it will be done voluntarily (the whole payment_info is optional). The real check to make sure `supported` tokens are used as payment is done in the gateway by the relayer.